adm/unix/compileunix.py

```diff
@@ -36,7 +36,9 @@ def gcc(exeFile, srcFiles, includes, libs, defs):
 
   # defines (tuples)
   for d in defs:
-    cmd += " -D" + d[0] + "=" + d[1]
+    cmd += " -D" + d[0]
+    if d[1] is not None:
```

Comment:
better to check the length of d here:
`
  if len(d) > 1 and d[1] is not None: 
`
this will allow user define defs like this: 
`
defs = [("FOO"), ("BAR", "1")] 
`

Or we can check if d is a tuple just at the beginning of the loop, that will allow user to just define it like this: 
`
defs = ["FOO", ("BAR", "1")] 
`

Reply:
len(d) will not work either. Because ("FOO") will be a string, not a tuple. We have to check for a type of d. I'll fix that.